src/ui/public/vis/response_handlers/legacy_table.js

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import get from 'lodash.get';
+import moment from 'moment';
+import AggConfigResult from '../../vis/agg_config_result';
+
+function keyRowsByValue(rows, key) {
+  return rows.reduce((acc, row) => {
+    const { [key]: splitValue, ...rest } = row;
+    acc[splitValue] = [...(acc[splitValue] || []), rest];
+    return acc;
+  }, {});
+}
+
+function formatSplitFieldValue(aggConfig, value) {
+  if (get(aggConfig, 'params.field.type') === 'date') {
+    return moment(parseInt(value)).format('YYYY-MM-DD');
+  }
+  return value;
+}
+
+export function legacyTableResponseHandler() {
+  return function handler(table) {
+    return new Promise(resolve => {
+      const splitTable = (columns, rows, $parent) => {
+        const splitColumn = columns.find(column => get(column, 'aggConfig.schema.name') === 'split');
+
+        if (!splitColumn) {
+          return [{
+            $parent,
+            columns: columns.map(column => ({ title: column.name, ...column })),
+            rows: rows.map(row => {
+              return columns.map(column => {
+                return new AggConfigResult(column.aggConfig, $parent, row[column.id], row[column.id]);
+              });
+            }),
+            aggConfig: (column) => column.aggConfig,
+          }];
+        }
+
+        const splitColumnIndex = columns.findIndex(column => column.id === splitColumn.id);
+        const rowsByColumnValue = keyRowsByValue(rows, splitColumn.id);
+        const filteredColumns = columns
+          .filter((column, i) => i !== splitColumnIndex)
```

Comment:
I haven't included any of the `isSplitMetric` logic [used in the legacy response handler](https://github.com/elastic/kibana/blob/master/src/ui/public/vis/response_handlers/legacy.js#L63) because it did not appear to be necessary for the tables -- but please correct me if I'm missing something.

Reply:
i think this might not be the case ....
if we have metrics at all levels, the tabified table will look like: (for two nested term aggs, first one on os, second one on countr)
```
os  coiunt  country  count
win xp  230 CN  200
win xp 230 US 30
linux 120 CN 100
linux 120 CN 20
```

lets say `os` is the split column .... we should filter out that one + all the matching metrics (so all the metrics until the next bucket column)

Reply:
@timroes and I were just testing and we *think* this isn't an issue based on our understanding (it seemed to behave consistently with both legacy and legacy_table handlers), but I will double check